### PR TITLE
Merge sort review

### PR DESCRIPTION
**Pull request by adelinami**

### CODE REVIEW

File: mergesort/mergesort.py [hunk 1]
```diff
@@ -1,23 +1,29 @@
...
     Returns:
         c: the created succesion using merge sort
     """
 
     c=[]
     i=0
     j=0
-    while i<n and j<m :
+    
+    if len(a)==len(b)==0:
```
Comment (rcristian) [inline mergesort/mergesort.py]: this case is covered by the following two; remove the statement

File: mergesort/mergesort.py [hunk 2]
```diff
@@ -1,23 +1,29 @@
...
+        return c
+    if len(a)==0: 
+        return b
+    if len(b)==0:
+        return a
+    
+    while i<len(a) and j<len(b) :
```
Comment (rcristian) [inline mergesort/mergesort.py]: len(a), len(b) get computed every time the check for the 'while' loop is performed; extract them in a var before the cycle

File: tests/test_mergesort.py [hunk 3]
```diff
@@ -7,13 +7,21 @@
...
+    c = merge_sort([1,2,3], [4,9])
     assert c==[1,2,3,4,9]
+    
+def test_corner_case_a():
+    c = merge_sort([], [1,2,3])
+    assert c==[1,2,3]
 
-def test_one_no_elements():
-    c = merge_sort([1,2,3], [], 3, 0)
+def test_corner_case_b():
```
Comment (rcristian) [inline tests/test_mergesort.py]: naming: 'test_second_list_empty'

File: tests/test_mergesort.py [hunk 4]
```diff
@@ -7,13 +7,21 @@
...
-    c = merge_sort([1,2,3], [2,3,4,9], 3, 4)
+    c = merge_sort([1,2,3], [2,3,4,9])
     assert c==[1,2,2,3,3,4,9]
 
 def test_happy_flow2():
-    c = merge_sort([1,2,3], [4,9], 3, 2)
+    c = merge_sort([1,2,3], [4,9])
     assert c==[1,2,3,4,9]
+    
+def test_corner_case_a():
```
Comment (rcristian) [inline tests/test_mergesort.py]: naming: 'test_first_list_empty'

File: tests/test_mergesort.py [hunk 5]
```diff
@@ -7,13 +7,21 @@
 
 
 def test_happy_flow():
-    c = merge_sort([1,2,3], [2,3,4,9], 3, 4)
+    c = merge_sort([1,2,3], [2,3,4,9])
     assert c==[1,2,2,3,3,4,9]
 
 def test_happy_flow2():
```
Comment (rcristian) [inline tests/test_mergesort.py]: no need to have two 'happy flows' - there's an infinite number of them, can't check all; focus on the corner cases

!!! tests are documentation, they explain the 'contract' - what you wrote in the docstring

File: mergesort/mergesort.py [hunk 6]
```diff
@@ -1,22 +1,27 @@
...
-        n: the number of elements in a
-        m: the number of elements in b
     Returns:
         c: the created succesion using merge sort
     """
 
     c=[]
     i=0
     j=0
+    n=len(a)
```
Comment (rcristian) [inline mergesort/mergesort.py]: naming: you're running out of letters for vars ;) a_size = len(a).. b_size = len(b).. "tell a story"
